### PR TITLE
Hibernation

### PR DESCRIPTION
**Pull request by storyaddict**
This PR Does the following: 

Takes the excellent process documentation created by @thomas-zimmerman here: https://github.com/pop-os/pop/issues/1966 

And expands/formats the content into a support page article.

This article describes how to enable hibernation on encrypted Pop installs.

### CODE REVIEW

**Review by ahoneybun — DISMISSED** [review]

File: content/enable-hibernation.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
+---
+title: Enable Hibernation (Suspend to Disk)
+description: >
+    How to partition a drive and set up hibernation in Pop!_OS.
```
[the remark this answers is not in the document]
Comment (storyaddict) [reply content/enable-hibernation.md:4]: This is going to be changed in a commit momentarily. Good note!

File: content/enable-hibernation.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+However, these limitations aside, hibernation is an often requested feature for Pop\!_OS. The purpose of this article is to provide steps for enabling hibernation for any users' that would like it.
+
+## Enable Hibernation on an Encrypted Drive
+
+1. Boot into [Pop!_OS Recovery](https://support.system76.com/articles/pop-recovery)
+
+    Turn your computer off, then turn it back on and hold down the `SPACE` bar immediately. In the menu that appears, select `PopOS Recovery`, and let it boot.
+
+    Once it boots, close out of the installation window or choose `try demo mode` (be sure not to choose any install or repair options, as this could result in data loss. It will also mount the drive and make the next steps harder).
```
[the remark this answers is not in the document]
Comment (storyaddict) [reply content/enable-hibernation.md:40]: Fixed capitalization (commit incoming).

File: content/enable-hibernation.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+    ![gparted](/images/hibernation/gparted.png)
+
+    Open the encrypted partition using these Terminal commands:
+
+    | **SATA Drives**                                    | **NVMe Drives**                                   |
+    |:--------------------------------------------------:|:-------------------------------------------------:|
+    | `sudo cryptsetup luksOpen /dev/sda3 cryptdata`       | `sudo cryptsetup luksOpen /dev/nvme0n1p3 cryptdata` |
+
+    > **NOTE:** In this example, the partition is called `/dev/vda3`. Adjust the drive name to your case.
```
[the remark this answers is not in the document]
Comment (storyaddict) [reply content/enable-hibernation.md:68]: Clarified this statement.

File: content/enable-hibernation.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+8. Add a resume target for the kernel:
+
+    Using the UUID retrieved from `fstab` run this command:
+
+    ```bash
+    sudo kernelstub -a "resume=UUID=xxxxxxxx-xxxx-xxxx-xxxxxxxx"
+    ```
+
+    > **NOTE:** Replace the UUID `xxx...` with the ID of your new `swap` volume.
```
[the remark this answers is not in the document]
Comment (storyaddict) [reply content/enable-hibernation.md]: Fixed.

File: content/enable-hibernation.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+---
+
+## State of Hibernation on Pop!_OS
+
+Hibernation is not currently enabled on Pop!_OS by default. The factors for this choice are:
+
+- Pop!_OS' Default Partition Layout
+- Use of a non-persistent encryption key for SWAP.
+- Additional drive I/O used on Solid State Drives (SSDs)
+- Depending on alloted RAM in a system, and size of NVMe/SSD drive(s), hibernation could add notable delay on resuming a hibernation session and wear out the OS drive prematurely.
```
Comment (thomas-zimmerman) [inline content/enable-hibernation.md:30]: Drop wear comment as it's part of the item above; size of swap space used should be a single item and delay should be it's own item.

File: content/enable-hibernation.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+twitterImage: /_social/article
+
+hidden: false
+section: software
+tableOfContents: true
+---
+
+## State of Hibernation on Pop!_OS
+
+Hibernation is not currently enabled on Pop!_OS by default. The factors for this choice are:
```
Comment (thomas-zimmerman) [inline content/enable-hibernation.md:25]: The factor for this choice are >> There are several limitations that keep this from being a default:

File: content/enable-hibernation.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+    Click on the encrypted `luks` partition and select the "Resize/Move" button. Use the GUI drag tools, or number fields to make sure the OS partition extends to the end of the drive (after `EFI` and `recovery`).
+
+    ![gparted](/images/hibernation/gparted.png)
+
+    Open the encrypted partition using these Terminal commands:
+
+    | **SATA Drives**                                    | **NVMe Drives**                                   |
+    |:--------------------------------------------------:|:-------------------------------------------------:|
+    | `sudo cryptsetup luksOpen /dev/sda3 cryptdata`       | `sudo cryptsetup luksOpen /dev/nvme0n1p3 cryptdata` |
```
Comment (thomas-zimmerman) [inline content/enable-hibernation.md:66]: change to terminal output: eg >> ```sudo cryptsetup luksOpen /dev/sda3 cryptdata```

File: content/enable-hibernation.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+    sudo kernelstub -a "resume=UUID=xxxxxxxx-xxxx-xxxx-xxxxxxxx"
+    ```
+
+    **NOTE:** Replace everything after `UUID=` with the ID of your new `swap` volume.
+
+    ![blkid](/images/hibernation/blkid.png)
+
+    The system is now ready to suspend to, and resume from disk.
+
+    You can test if hibernation works on your system by running:
```
Comment (thomas-zimmerman) [inline content/enable-hibernation.md:127]: ```You can test if hibernation works by booting back into your install and running```

**Review by thomas-zimmerman — DISMISSED** [review]
See changes requested inline

File: content/enable-hibernation.md (hunk 5 above)
Comment (storyaddict) [reply content/enable-hibernation.md:30]: Wear has been removed, delay retained.

File: content/enable-hibernation.md (hunk 7 above)
Comment (storyaddict) [reply content/enable-hibernation.md:66]: Added the ``` ticks.

File: content/enable-hibernation.md (hunk 8 above)
Comment (storyaddict) [reply content/enable-hibernation.md]: Edited.

File: content/enable-hibernation.md (hunk 6 above)
Comment (storyaddict) [reply content/enable-hibernation.md]: Fixed.

**Review by storyaddict — COMMENTED** [review]
Good change, thanks Alan! :)

**Review by Schykle — APPROVED** [review]

**Pull request merged**